### PR TITLE
Draft: Initial attempt at macos support

### PR DESCRIPTION
**Pull request by wlcx**
Initial go at adding support for other platforms. 

#33

### CODE REVIEW

**Comment by tastycode** [comment]
This appears to work. When i check out `wlcx:other-platforms` and run `nix develop ".#esp32-idf"`. I get a shell, which is farther than i got after 2 hours of trying to get the repo's shell.nix example to work as a flake. 

i originally went down the route of esp-idf after seeing some GUI mentioned in  the nimBLE repo. I was curious about it then I saw how involved it was to setup esp-idf so i thought there might be a nix setup to make it easier. The program i was trying to run is "menuconfig", but when i try to run it, i get 
```
bash-5.2$ menuconfig
/nix/store/w2ic8d5f5cw2kj03dg5kjqys0sfz0acr-python3.11-kconfiglib-14.1.0/bin/menuconfig: [Errno 2] No such file or directory: 'Kconfig'
```

seems like you did great work though! I hadn't gotten a nix command to run succesfully in hours until i tried your fork.

**Comment by wlcx** [comment]
You might wanting `idf.py menuconfig` ?

File: pkgs/esp-idf/tools.nix [hunk 1]
```diff
@@ -81,7 +87,15 @@ let
 
       phases = [ "unpackPhase" "installPhase" ];
 
-      installPhase = ''
+      installPhase = let
+        wrapCmd = if system == "linux-x86_64" then
+        ''
+          makeWrapper ${fhsEnv}/bin/${pname}-env $FILE_PATH --add-flags "$FILE_PATH-unwrapped" ${lib.strings.concatStringsSep " " exportVarsWrapperArgsList}
```
Comment (mirrexagon) [inline pkgs/esp-idf/tools.nix:93]: I actually don't know if this whole thing is required, given what you've done for other platforms. For now I think it's fine, as long as it works for all the supported platforms.

**Review by mirrexagon — APPROVED** [review]
Thank you for this! I can't test this on other platforms, but it doesn't seem to break anything for me on NixOS x86_64, and it looks good.

File: pkgs/esp-idf/tools.nix (hunk 1 above)
Comment (wlcx) [reply pkgs/esp-idf/tools.nix:93]: Yeah this was a bit of a hack, I didn't want to disturb what was already there in case it was load bearing :)

**Pull request merged**